File: CustomShare/src/main/java/abhishekti/appyhigh/customshare/CustomShare.java
package abhishekti.appyhigh.customshare;

import android.app.Activity;
import android.content.Intent;
import android.content.pm.ApplicationInfo;
import android.content.pm.PackageManager;
import android.content.pm.ResolveInfo;
import android.os.Build;
import android.util.Log;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.fragment.app.FragmentManager;

import java.lang.ref.WeakReference;
import java.util.List;

import abhishekti.appyhigh.customshare.models.Config;
import abhishekti.appyhigh.customshare.ui.CustomShareBottomSheet;

/**
 * Created by Abhishek Tiwari on 18-01-2021.
 */
public final class CustomShare {
    private static final String TAG = "CustomShare";
    private final WeakReference<Activity> mActivity;
    private final FragmentManager fragmentManager;

    private CustomShare(Activity activity, FragmentManager manager){
        this.mActivity = new WeakReference<>(activity);
        this.fragmentManager = manager;
    }

    public static CustomShare getInstance(Activity activity, FragmentManager fragmentManager){
        return new CustomShare(activity, fragmentManager);
    }

    public void show(Config config){
        CustomShareBottomSheet bottomSheet = new CustomShareBottomSheet(config);
        bottomSheet.show(fragmentManager, "custom_share");
    }

    public Builder Builder(){
        return new Builder(this);
    }

    @Nullable
    Activity getActivity(){
        return mActivity.get();
    }

}

File: CustomShare/src/main/java/abhishekti/appyhigh/customshare/CustomShare.kt
package abhishekti.appyhigh.customshare

import abhishekti.appyhigh.customshare.models.Config
import abhishekti.appyhigh.customshare.ui.CustomShareBottomSheet
import android.app.Activity
import androidx.fragment.app.FragmentManager
import java.lang.ref.WeakReference

/**
 * Created by Abhishek Tiwari on 02-03-2021.
 */
public class CustomShare private constructor(activity: Activity, manager: FragmentManager) {
    private val mActivity: WeakReference<Activity> = WeakReference(activity)
    private val fragmentManager: FragmentManager = manager

    fun show(config: Config?) {
        val bottomSheet = CustomShareBottomSheet(config)
        bottomSheet.show(fragmentManager, "custom_share")
    }

    fun Builder(): Builder {
        return Builder(this)
    }

    fun getActivity(): Activity? {
        return mActivity.get()
    }

    companion object {
        private const val TAG = "CustomShare"
        fun getInstance(activity: Activity, fragmentManager: FragmentManager): CustomShare {
            return CustomShare(activity, fragmentManager)
        }
    }

}
